device/device_flow.go

```diff
@@ -54,12 +54,29 @@ type CodeResponse struct {
 	Interval int
 }
 
+// Function signature for setting additional form values.
+type AuthRequestFn func(*url.Values)
+
+func WithAudience(audience string) AuthRequestFn {
+	return func(values *url.Values) {
+		if audience != "" {
+			values.Add("audience", audience)
+		}
+	}
+}
+
 // RequestCode initiates the authorization flow by requesting a code from uri.
-func RequestCode(c httpClient, uri string, clientID string, scopes []string) (*CodeResponse, error) {
-	resp, err := api.PostForm(c, uri, url.Values{
+func RequestCode(c httpClient, uri string, clientID string, scopes []string, extra ...AuthRequestFn) (*CodeResponse, error) {
+	values := url.Values{
 		"client_id": {clientID},
 		"scope":     {strings.Join(scopes, " ")},
-	})
+	}
+
+	for _, fn := range extra {
+		fn(&values)
+	}
+
+	resp, err := api.PostForm(c, uri, values)
```

Comment:
@jtmcg Thanks for the review. I have made the change. Updated the variable name to `optionalRequestParams`. I feel it is intention revealing while hiding the internal implementation detail of using `url.Value`. Let me know what you think about it?

Also I choose this approach because I wanted to avoid breaking the API contract since `RequestCode` is public. It would be harder to justify a major version upgrade for such a simple enhancement.